internals/daemon/api_exec.go

```diff
@@ -73,13 +75,29 @@ func v1PostExec(c *Command, req *http.Request, _ *userState) Response {
 		return statusBadRequest("%v", err)
 	}
 
+	// Inherit the pebble daemon environment.
+	environment := make(map[string]string)
+	for _, kv := range os.Environ() {
+		parts := strings.SplitN(kv, "=", 2)
+		key := parts[0]
+		val := ""
+		if len(parts) == 2 {
+			val = parts[1]
+		}
+		environment[key] = val
+	}
+	for key, val := range payload.Environment {
```

Comment:
wouldn't it be better to just expand payload.Environment whenever a key from os.Environ doesn't exist, instead of iterating over two maps?

Reply:
`payload.Environment` may be `nil`. In that case, it will error out; it's safer to just iterate over it.